쪽지함 조회 및 회원 정보 dto 생성·수정

dldmldlsy opened the pull request:
## 관련 이슈
-  Close #59 

## 작업 사항
- 

## 참고 자료
- 참고자료 (링크 등)

## 체크리스트
- [x]  PR의 제목은 팀 내 규칙을 준수하여 알맞게 작성하였는가?
- [x]  Merge 하는 브랜치가 올바른가? (`main` branch에 실수로 PR 생성 금지)
- [x]  팀의 코딩 컨벤션을 준수하는가?
- [ ]  PR과 관련없는 변경사항이 들어가지는 않았는가?
- [x]  내 코드에 대한 자기 검토가 되었는가?
- [x]  Reviewers, Assignees, Lables, Project, Milestone은 적절하게 선택하였는가?
- [ ]  관련한 issue를 닫아야 하는지 점검해보고 적용했는가?

dldmldlsy committed:
fix: idnotfoundexception 관련 충돌 해결

dldmldlsy committed:
feat: #59 쪽지 1개 정보 dto, 회원 대표적인 프로필 정보 dto 생성

dldmldlsy committed:
feat: #59 쪽지함 조회 api controller 구현

dldmldlsy committed:
#feat: 회원 전체 정보 dto 생성 후 에러 수정

항목별 총합, 리뷰 수 필드 추가/ 리뷰 개수 조회 메서드 추가, 모임관련 에러 수정, 리뷰idnotfound 에러처리 코드 수정

dldmldlsy committed:
feat: #59 쪽지함 service 구현중

service 구현중, 에러 수정

Wo-ogie commented on src/main/java/com/umc/withme/controller/MessageController.java:
```diff
@@ -0,0 +1,69 @@
+
+    /**
+     * 쪽지 생성 API
+     *
+     * @param messageCreateRequest 쪽지 생성 요청 데이터 (쪽지 내용)
+     * @param principle
+     * @param receiverId 쪽지 받는 회원 id(pk)
+     * @param meetId 관련 모집글 id(pk)
+     * @return 쪽지 생성 응답 데이터 (생성된 쪽지 아이디)
+     */
+
+    @Operation(summary = "쪽지 생성",
+               description = "<p> 회원이 쪽지 1개를 생성합니다:: <code>meetId</code> - 관련 모집글, <code>receiverId</code> - 받는 회원, " +
+                             "<code>messageCreateRequest</code>의 <code>content</code></p> - 쪽지 내용" )
+    @PostMapping("/messages")
+    public ResponseEntity<DataResponse<MessageCreateResponse>> createMessage(
+            @Valid@RequestBody MessageCreateRequest messageCreateRequest,
+            @AuthenticationPrincipal WithMeAppPrinciple principle, @RequestParam Long receiverId,
+            @RequestParam Long meetId){
+
+        Long messageId = messageService.createMessage(principle.getMemberId(), receiverId, meetId, messageCreateRequest);
+        MessageCreateResponse messageCreateResponse = MessageCreateResponse.of(messageId);
+
+        return ResponseEntity.status(HttpStatus.CREATED).body(new DataResponse<>(messageCreateResponse));
+    }
+
+    /**
+     *
+     * @param principle
+     * @return 쪽지함 조회 응답 데이터 (회원 프로필 정보, 쪽지 내용, 작성시간)
+     */
+    @Operation(summary = "쪽지함 조회", description = "<p> 회원이 자신의 쪽지함을 조회합니다</p>")
+    @GetMapping("/messages")
+    public ResponseEntity<DataResponse<List<MessageInfoResponse>>> getMessageList(
+            @Parameter(hidden = true) @AuthenticationPrincipal WithMeAppPrinciple principle){
+
+        List<MessageInfoResponse> messageInfoResponseList = messageService.getMessageList(principle.getMemberId());
+
+        return ResponseEntity.status(HttpStatus.OK).body(new DataResponse<>(messageInfoResponseList));
+    }
```
저희 팀 문서에 보시면, controller layer는 Swagger 명세가 있으니까 주석 굳이 안달아도 된다고 했었습니다. 
지워주셔도 괜찮을 거 같아요 😊

dldmldlsy replied:
아하 넵!!!!

Wo-ogie committed:
docs: 모임 조회 API 명세 통일 및 불필요한 주석 제거

Wo-ogie committed:
refactor: #59 받은 리뷰 수 조회 및 신뢰 점수 계산 로직 수정

기존 코드에서는 dto 각 메서드의 역할과 parameter가 조금 어색한 부분이 있었다. 이에 다음과 같은 내용들을 적용하여 수정하였음.

1. `Member`에 "받은 리뷰 수" 필드 추가 및 `@Formula`를 활용하여 자체적으로 받은 리뷰 수를 조회할 수 있도록 하였음.
2. 1번의 변경사항으로 더 이상 필요 없어진 `ReviewService`의 받은 리뷰 수 조회 메서드 삭제
3. 신뢰 점수 계산 로직을 `TotalPoint` class에 위임.

Wo-ogie committed:
refactor: #59 받은 리뷰 수의 type을 primitive type(int)에서 Integer로 변경

hyeonnkang committed:
feat: #59 쪽지 생성 기능 구현 및 테스트

- 기존 API에 principle 내용이 노출 되는 문제 해결
- MessageController 에 access token 코드 추가
- MessageCreateRequest에 기본 생성자 추가

위 변경사항으로 쪽지 생성 기능이 작동되는 것을 스웨거 테스트 완료하였습니다.

hyeonnkang committed:
feat: #59 쪽지함 조회 기능 구현 및 테스트

- MessageInfoResponse의 createAt -> createdAt으로 네이밍 변경
- 쪽지함 조회 쿼리를 JPQL로 작성하는 데에 어려움을 겪어 우선 네이티브 쿼리로 작성하였습니다. 추후에 JPQL로 변경하는게 좋을 것 같아요.
- MemberShortInfoResponse의 of 메소드 내부에서 받은 후기 갯수가 0일때는 나누기 연산을 하지 않도록 if문 검증부분을 하나 추가했습니다.

hyeonnkang committed:
Merge branch 'develop' into feature/#59-messages-list-up

hyeonnkang committed:
feat: #59 쪽지함 조회 쿼리 수정

쪽지함 조회 시 오류를 발견하여 쿼리를 일부 수정했다.
기존에는 모임의 id 로만 구분해서 최신 쪽지를 조회해왔는데, 작성자의 id로도 그루핑 하는 것으로 변경해주었다.

dldmldlsy committed:
fix: #59 MessageController에 주석 삭제

swagger명세서에 api 설명이 있기 때문에 불필요하여 삭제

dldmldlsy committed:
feat: origin pull받기

Wo-ogie committed:
Merge branch 'origin/feature/#59-messages-list-up' into feature/#59-refactor-calculate-trust-point

충돌 해결 후 merge.

Wo-ogie committed:
Merge pull request #73 from UMC-WithMe/feature/#59-refactor-calculate-trust-point

받은 리뷰 수 조회 및 신뢰 점수 계산 로직 변경

hyeonnkang committed:
feat: #59 쪽지함 조회 기능 구현 및 테스트

mysql 을 jpql로 변경하는 것에 어려움이 있어 네이티브 쿼리를 사용하여 쪽지함 조회 구현했습니다.

스웨거 테스트는 성공했으나, 제 카톡 액세스 토큰으로 로그인하고 나머지 멤버는 임의로 넣어 테스트 진행했습니다.
별 문제 없을 듯 하지만 추후 실제 카톡 액세스 토큰으로 2명 이상의 멤버가 로그인했을 때 잘 작동하는 지는 확인해봐야 될 것 같아요

dldmldlsy committed:
docs: #59 erd에쪽지 채팅방 entity(chatroom) 추가

dldmldlsy committed:
Merge branch 'feature/#59-messages-list-up' of https://github.com/UMC-WithMe/WithMe_Back into feature/#59-messages-list-up

dldmldlsy committed:
feat: #59 develop merge 및충돌 해결

dldmldlsy committed:
Merge branch 'develop' into feature/#59-messages-list-up

dldmldlsy committed:
feat: 모임해제 api merge 후 충돌 해결

dldmldlsy committed:
Merge branch 'develop' into feature/#59-messages-list-up

dldmldlsy committed:
memberShortDto에 회원 프로필 추가
